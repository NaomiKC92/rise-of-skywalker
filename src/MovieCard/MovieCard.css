* {
  box-sizing: border-box;
  font-family: 'PT Sans Narrow', sans-serif;
}

.movie-card {
  background-color: #04243D;
  border: 8px ridge #FFF;
  text-align: center;
  display: flex;
  flex-direction: column;
  font-size: 21px;
  justify-content: space-between;
  width: 18%;
  height: 300px;
  margin: 15px;
  padding: 1.5rem;
  color: #FFF;
}

h2 {
  font-family: 'Squada One', cursive;
  letter-spacing: 1px;
  font-size: 30px;
}

.movie-card-button {
  background: #FFF;
  border: 2px solid #fff;
  border-radius: 3px;
  color: #04243D;
  cursor: pointer;
  display: inline-block;
  font-size: 1.2rem;
  padding: 0.4rem 1.3rem;
  transition: opacity 0.2s ease-in;
  font-weight: bold; 
}

.movie-card-button :hover {
  cursor: pointer;
}
