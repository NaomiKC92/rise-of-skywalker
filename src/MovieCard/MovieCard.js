import React from 'react';
import PropTypes from 'prop-types';
import './MovieCard.css';

const MovieCard = ({
  title,
  episode_id,
  release_date,
  setupCharacters,
  movieIndex
}) => {
  return (
    <div className='card'>
      <h2>{title}</h2>
      <h3>Episode: {episode_id}</h3>
      <h3>Release: {release_date}</h3>
      <button
        id={episode_id}
        className='button'
        type='button'
        onClick={event => {
          console.log(event.target.id);
          console.log(movieIndex);
          setupCharacters(movieIndex + 1);
        }}>
        View Characters
      </button>
    </div>
  );
};

export default MovieCard;

MovieCard.propTypes = {
  title: PropTypes.string.isRequired,
  episode_id: PropTypes.number.isRequired,
  release_date: PropTypes.string.isRequired
};
