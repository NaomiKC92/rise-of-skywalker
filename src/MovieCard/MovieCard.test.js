import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MOVIE CARD', () => {
  let wrapper;

  const mockTitle = 'Rise of Skywalker';
  const mockEpisodeID = 8;
  const mockReleaseDate = '2019';
  const mockSetupCharacters = jest.fn();
  const mockUpdateCrawling = jest.fn();
  const mockMovieIndex = 7;
  const mockID = 8;

  beforeEach(() => {
    wrapper = shallow(
      <MovieCard
        title={mockTitle}
        episode_id={mockEpisodeID}
        release_date={mockReleaseDate}
        setupCharacters={mockSetupCharacters}
        updateCrawling={mockUpdateCrawling}
        movieIndex={mockMovieIndex}
        id={mockID}
      />
    );
  });

  it('should match the MovieCard Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
