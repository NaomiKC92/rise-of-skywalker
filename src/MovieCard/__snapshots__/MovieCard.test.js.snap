// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHARACTER CONTAINER should match the MovieCard Component Snapshot 1`] = `undefined`;

exports[`MOVIE CARD should match the MovieCard Component Snapshot 1`] = `
<div
  className="card"
>
  <h2>
    Rise of Skywalker
  </h2>
  <h3>
    Episode: 
    8
  </h3>
  <h3>
    Release: 
    2019
  </h3>
  <Link
    to="/movies/8"
  >
    <button
      className="button"
      id={8}
      onClick={[Function]}
      type="button"
    >
      View Characters
    </button>
  </Link>
</div>
`;
